data-prepper-plugins/mutate-event-processors/README.md

```diff
@@ -246,6 +246,101 @@ and the type conversion processor will change it to the following output, where
+```
+
+If we set a `target`:
+```yaml
+    - list_to_map:
+        key: "name"
+        source: "mylist"
+        target: "mymap"
+        value_key: "value"
+        flatten: true
+```
+the generated map will be put under the target key:
+```json
+{"mylist":[{"name":"a","value":"val-a"},{"name":"b","value":"val-b1"},{"name":"b","value":"val-b2"},{"name":"c","value":"val-c"}],"mymap":{"a":"val-a","b":"val-b1","c":"val-c"}}
+```
+
+If we do not specify a `value_key`:
+```yaml
+    - list_to_map:
+        key: "name"
+        source: "mylist"
+        flatten: true
+```
+the values of the map will be original objects from the source list:
+```json
+{"mylist":[{"name":"a","value":"val-a"},{"name":"b","value":"val-b1"},{"name":"b","value":"val-b2"},{"name":"c","value":"val-c"}],"a":{"name":"a","value":"val-a"},"b":{"name":"b","value":"val-b1"},"c":{"name":"c","value":"val-c"}}
+```
+
+If `flatten` option is not set or set to false:
+```yaml
+    - list_to_map:
+        key: "name"
+        source: "mylist"
+        value_key: "value"
+        flatten: false
+```
+the values of the map will be lists, and some entries may have more than one element in their values:
+```json
+{"mylist":[{"name":"a","value":"val-a"},{"name":"b","value":"val-b1"},{"name":"b","value":"val-b2"},{"name":"c","value":"val-c"}],"a":["val-a"],"b":["val-b1","val-b2"],"c":["val-c"]}
+```
+
+If we specify `flattened_element` as "last":
+```yaml
+    - list_to_map:
+        key: "name"
+        source: "mylist"
+        value_key: "value"
+        flatten: true
+        flattened_element: "last"
+```
+the last element will be kept:
+```json
+{"mylist":[{"name":"a","value":"val-a"},{"name":"b","value":"val-b1"},{"name":"b","value":"val-b2"},{"name":"c","value":"val-c"}],"a":"val-a","b":"val-b2","c":"val-c"}
+```
+
+### Configuration
+* `key` - (required) - The key of the fields that will be extracted as keys in the generated map
+* `source` - (required) - The key of the source list
+* `target` - (optional) - The key of the field that will hold the generated map. If not specified, the generated map will be put in the root.
+* `value_key` - (optional) - If specified, the values of the given `value_key` in the objects of the source list will be extracted and put into the values of the generated map; otherwise, original objects in the source list will be put into the values of the generated map.
+* `flatten` - (optional) - A boolean value, default to false. If it's false, the values in the generated map will be lists; if it's true, the lists will be flattened into single items.
+* `flattened_element` - (optional) - Valid options are "first" and "last", default is "first". This specifies which element, first one or last one, to keep if `flatten` option is true.
+
 
 ## Developer Guide
 This plugin is compatible with Java 14. See
```

Comment:
Is there supposed to be reference to guide here?

Reply:
Added the actual developer guide and plugin development guide.